libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -628,78 +683,37 @@ static IotMqttError_t _subscriptionCommon( IotMqttOperationType_t operation,
     {
         EMPTY_ELSE_MARKER;
     }
+    IOT_FUNCTION_CLEANUP_BEGIN();
+    IOT_FUNCTION_CLEANUP_END();
+}
 
-    /* Choose a subscription serialize function. */
-    if( operation == IOT_MQTT_SUBSCRIBE )
-    {
-        serializeSubscription = _IotMqtt_SerializeSubscribe;
-
-        #if IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES == 1
-            if( mqttConnection->pSerializer != NULL )
-            {
-                if( mqttConnection->pSerializer->serialize.subscribe != NULL )
-                {
-                    serializeSubscription = mqttConnection->pSerializer->serialize.subscribe;
-                }
-                else
-                {
-                    EMPTY_ELSE_MARKER;
-                }
-            }
-            else
-            {
-                EMPTY_ELSE_MARKER;
-            }
-        #endif /* if IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES == 1 */
-    }
-    else
-    {
-        serializeSubscription = _IotMqtt_SerializeUnsubscribe;
-
-        #if IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES == 1
-            if( mqttConnection->pSerializer != NULL )
-            {
-                if( mqttConnection->pSerializer->serialize.unsubscribe != NULL )
-                {
-                    serializeSubscription = mqttConnection->pSerializer->serialize.unsubscribe;
-                }
-                else
-                {
-                    EMPTY_ELSE_MARKER;
-                }
-            }
-            else
-            {
-                EMPTY_ELSE_MARKER;
-            }
-        #endif /* if IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES == 1 */
-    }
+/*-----------------------------------------------------------*/
 
-    /* Remove the MQTT subscription list for an UNSUBSCRIBE. */
-    if( operation == IOT_MQTT_UNSUBSCRIBE )
-    {
-        _IotMqtt_RemoveSubscriptionByTopicFilter( mqttConnection,
-                                                  pSubscriptionList,
-                                                  subscriptionCount );
-    }
-    else
-    {
-        EMPTY_ELSE_MARKER;
-    }
+static IotMqttError_t _subscriptionCreateAndSerialize( IotMqttOperationType_t operation,
+                                           IotMqttConnection_t mqttConnection,
+										   _subscribeSerializer_t serializeSubscription,
+                                           const IotMqttSubscription_t * pSubscriptionList,
+                                           size_t subscriptionCount,
+                                           uint32_t flags,
+                                           const IotMqttCallbackInfo_t * pCallbackInfo,
+                                           _mqttOperation_t ** ppSubscriptionOperation )
```

Comment:
Tabs &rarr; spaces, and more spaces to uncrustify and line up parameters. Do we have a tool set up like uncrustify as we do for the afr repo?

Reply:
Fixed.  I'm not aware of an uncrustify git-hook for this repo but that would be a good addition.

Reply:
My vim's still highlighting tab characters on lines 1488-1489, 1498-1503, 1559-1560, and 1574-1579. Can you replace those too?

Reply:
Good catch, should be fixed in latest revision